src/Interpreters/ClusterProxy/executeQuery.cpp

```diff
@@ -512,6 +514,17 @@ static ContextMutablePtr updateContextForParallelReplicas(const LoggerPtr & logg
         /// disable hedged connections -> parallel replicas uses own logic to choose replicas
         context_mutable->setSetting("use_hedged_requests", Field{false});
     }
+
+    /// If parallel replicas are scoped to a specific shard, The local plan for parallel replicas cannot be generated.
+    /// And projections may need to be read from the local plan in certain scenarios, such as min-max projection. So we disable it here.
+    if (shard_num > 0 && settings[Setting::parallel_replicas_support_projection].value)
```

Comment:
The test doesn't have any projections involved. It looks the error should still appear if parallel_replicas_support_projection is disabled.

Reply:
Indeed, I had set the wrong ORDER BY key. Now fixed.

Reply:
AFAIK, in addition to explicitly defined projections, when optimizing aggregate projections (e.g., count/min/max), virtual projections are generated for trivial optimizations. These rely solely on index analysis and do not require reading part data. In such cases, `ReadFromMergeTree` is replaced with `ReadFromPreparedSource`.

To prevent data duplication caused by all replicas executing `ReadFromPreparedSource`, only the local plan executes `ReadFromPreparedSource`. In the example of this issue, no local plan was generated, leading to incorrect results.

If we disable `parallel_replicas_support_projection`, the projection optimization is skipped, and `ReadFromMergeTree` is executed to read part data, which avoids the issue.